### PR TITLE
Add background-with-triangles processor

### LINKED ISSUE #36: Background triangles
## Description
In most of the pages of the web, there are some triangles in the background for styling them.
Frontity_Framework_Landing_Play_Mode_-_InVision-5

While some of the properties like colors or sizes while be the same for all the triangles, there are some that can be different depending on the page:
* position: They can be placed in `left`, `right`, `both-sides` or `none`.
* top: The position from the top may vary depending the page.
* top-triangle-opacity: The opacity of the triangle on the top level may vary as well.

In order to get these properties for each page, we are planning to pass them using Advanced Custom Fields and the REST API, so in each page we can find a field named `acf`:
https___wp_frontity_org_wp-json_wp_v2_pages_

Note that this is our first idea, feel free to modify it if you consider it's better other way.

## Possible solution
In the Frontity project, inside the `<Page />` component, we could include a new component to handle these background triangles. Apart from that, we should fetch the `acf` field to get the customizable properties for each page (homepage, about-us, contact-us...).

The component could be something similar to this (adapting it to the Invision real styles and valid React):
> Note that this code is not valid (nor the javascript or the css), is just to have an idea of the possible implementation.
```
const { position, top, top-triangle-opacity } = acf.background-triangles;

const BackgroundTriangles = ( ) => {
  <TrianglesContainer>
        <Triangle1 />
        <Triangle2 />
  </TrianglesContainer>
};
export default connect(BackgroundTriangles);

const TrianglesContainer = styled.div`
  position: absolute;
  top: {$top};
  display: block;
  width: 0px;
  align-items: center;
  transform: rotate(45deg);
`;
const BackgroundTriangle = styled.div`
  position: absolute;
  width: 817px;
  height: 817px;
`;
const Triangle1 = styled(BackgroundTriangle)`
  box-shadow: 0 0 14px 0 rgba(12, 17, 43, 0.03);
  z-index: 1000;
  left: 114px;
  top: -114px;
  background: white;
  opacity: 0.8;
`;
const Triangle2 = styled(BackgroundTriangle)`
  z-index: 2000;
  background: gray;
  opacity: {$opacity}
`;
```
There is a similar implementation in the footer of the current web at https://frontity.org, it could be useful as well.
## Where to check
In order to get the correct styles, we can take a look at Invision, at the homepage and the `about-us` page.
In order to check if it's working in the Frontity side, we've added different Advanced Custom Fields to the `/homepage` and the `/common-styles/` page.

### PR DESCRIPTION
**Pull request by iamuchejude**
Fixed #36

### CODE REVIEW

**Comment by frontibotito** [comment]
Deploy preview for _website_ ready!

Built with commit f90fac05cf692c4c7eca8b380b7d2733e09b0eab

https://frontity-org-76umgzpzf.now.sh

**Comment by SantosGuillamot** [comment]
@iamuchejude I've just seen that you're planning to add a processor to solve this. Our first idea was to add Advanced Custom Fields for each page including the custom properties you need as explained in [the issue](https://github.com/frontity/frontity.org/issues/36). This way, you could find them in the REST API response, and you could get them from Frontity state. You wouldn't need a processor with this approach I think.

Did you come up with a different solution?

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    <Container height={height}>
+      {position !== "both-sides" && position !== "none" ? (
+        <TriangleComp
+          topTriangleOpacity={topTriangleOpacity}
+          position={position}
+          top={top}
+        />
+      ) : null}
+
+      {position !== "both-sides" &&
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/components/background-with-triangles.tsx]: I think this logic is doing just the opposite, it should be `position === "both-sides"` right?

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  width: 750px;
+  background: linear-gradient(
+    135deg,
+    rgba(255, 255, 255, 0) 0%,
+    rgba(255, 255, 255, 0.8) 100%
+  );
+
+  ${(props) => css`
+    &.position-left {
+      transform: translate(-400px, ${props.top ? `${props.top}px` : "-52.5%"})
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/components/background-with-triangles.tsx]: I would change the translate top property in order to make the triangles start at the top if its property is `0px`. Something like this:
`transform: translate(-400px, calc(${props.top ? ${props.top}px : "0px"} - 360px))`

This way, if top property is 0, it matches directly the default design.

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        rotate(45deg);
+      transform-origin: left;
+    }
+
+    &.position-right {
+      transform: translate(400px, ${props.top ? `${props.top}px` : "-52.5%"})
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/components/background-with-triangles.tsx]: Same here: I would change the translate top property in order to make the triangles start at the top if its property is `0px`. Something like this:
`transform: translate(400px, calc(${props.top ? ${props.top}px : "0px"} - 360px))`

This way, if top property is 0, it matches directly the default design.

**Comment by SantosGuillamot** [comment]
Oh, I forgot to mention it in the review. We should add some property to hide it in mobile version.

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx (hunk 1 above)
Comment (iamuchejude) [reply packages/frontity-org-theme/src/components/background-with-triangles.tsx]: Oh yeah! I changed that to debug something. I'll fix it.

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    135deg,
+    rgba(255, 255, 255, 0) 0%,
+    rgba(255, 255, 255, 0.8) 100%
+  );
+
+  ${(props) => css`
+    &.position-left {
+      transform: translate(
+          -400px,
+          calc(${props.top ? `${props.top}px` : "0px"} - 35%)
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/components/background-with-triangles.tsx:90]: In order to match the default style ( the one in the about us page) when we pass 0px in the `top` property, I'd change the `35%` and use a bigger number. I've tested it an it should be `48%` more or less.

File: packages/frontity-org-theme/src/components/background-with-triangles.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        )
+        rotate(45deg);
+      transform-origin: left;
+    }
+
+    &.position-right {
+      transform: translate(
+          400px,
+          calc(${props.top ? `${props.top}px` : "0px"} - 35%)
```
Comment (SantosGuillamot) [inline packages/frontity-org-theme/src/components/background-with-triangles.tsx:99]: In order to match the default style ( the one in the about us page) when we pass 0px in the `top` property, I'd change the `35%` and use a bigger number. I've tested it an it should be `48%` more or less.

**Comment by iamuchejude** [comment]
Thanks @SantosGuillamot @DAreRodz

**Pull request merged**